server/src/test/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResourceTest.java

```diff
@@ -0,0 +1,103 @@
+package keywhiz.service.resources.automation.v2;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.UUID;
+import keywhiz.api.ApiDate;
+import keywhiz.api.model.AutomationClient;
+import keywhiz.api.model.Client;
+import keywhiz.jooq.tables.records.SecretsRecord;
+import keywhiz.log.AuditLog;
+import keywhiz.log.Event;
+import keywhiz.log.EventTag;
+import keywhiz.service.crypto.RowHmacGenerator;
+import keywhiz.test.TestDSLContexts;
+import org.jooq.DSLContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class BackfillRowHmacResourceTest {
+  @Mock
+  private RowHmacGenerator rowHmacGenerator;
+  @Mock
+  private AuditLog auditLog;
+
+  @Captor
+  private ArgumentCaptor<Event> eventCaptor;
+
+  @Before
+  public void before() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void backfillSecretHmacByNameHandlesNullOldHmac() {
+    SecretsRecord record = new SecretsRecord();
```

Comment:
Although technically not a possibility in the really-real world because null HMACs are no longer allowed by the database schema, this test arose after discovering that the ImmutableMap used to track the before-and-after HMACs in the audit log entry barfed if the old HMAC was null, which means it would fail to satisfy the actually-backfilling-a-null-HMAC use case.